pkg/controller/operatingsystemconfig/os-common/app/app.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	extcontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	controllercmd "github.com/gardener/gardener-extensions/pkg/controller/cmd"
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/os-common/actuator"
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/os-common/generator"
+	"github.com/spf13/cobra"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// NewControllerCommand creates a new command for running an OS controller.
+func NewControllerCommand(ctx context.Context, osName string, generator generator.Generator, options *controller.Options) *cobra.Command {
+	var (
+		restOpts = &controllercmd.RESTOptions{}
+		mgrOpts  = &controllercmd.ManagerOptions{
+			LeaderElection:          true,
+			LeaderElectionID:        controllercmd.LeaderElectionNameID(osName),
+			LeaderElectionNamespace: os.Getenv("LEADER_ELECTION_NAMESPACE"),
+		}
+		ctrlOpts = &controllercmd.ControllerOptions{
+			MaxConcurrentReconciles: 5,
+		}
+
+		aggOption = controllercmd.NewOptionAggregator(restOpts, mgrOpts, ctrlOpts)
+	)
+
+	cmd := &cobra.Command{
+		Use: "os-" + osName + "-controller-manager",
+
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := aggOption.Complete(); err != nil {
+				controllercmd.LogErrAndExit(err, "Error completing options")
+			}
+
+			mgr, err := manager.New(restOpts.Completed().Config, mgrOpts.Completed().Options())
+			if err != nil {
+				controllercmd.LogErrAndExit(err, "Could not instantiate manager")
+			}
+
+			if err := extcontroller.AddToScheme(mgr.GetScheme()); err != nil {
+				controllercmd.LogErrAndExit(err, "Could not update manager scheme")
+			}
+
+			ctrlOpts.Completed().Apply(options)
+
+			if err := actuator.Add(mgr, osName, generator); err != nil {
```

Comment:
An actuator should not be '`Add`ed' to a manager, only a controller.

Reply:
Agree. In the refactoring this function ended in the actuator package. Moved to a new 'oscommon' package,